include/container/sycl_iterator.h

```diff
@@ -35,25 +35,25 @@ class BufferIterator {
  public:
   using scalar_t = element_t;
   template <int dim = 1>
-  using buffer_t = cl::sycl::buffer<scalar_t, dim>;
-  using access_mode_t = cl::sycl::access::mode;
-  template <cl::sycl::access::mode acc_md_t =
-                cl::sycl::access::mode::read_write,
-            cl::sycl::access::target access_t =
-                cl::sycl::access::target::global_buffer,
-            cl::sycl::access::placeholder place_holder_t =
```

Comment:
`access::placeholder` is deprecated in 2020. I can see this might be more work to fix, and would be supportive of this being a different PR.

Reply:
I spend some time checking how to deal with it. 
SYCL2020 spec seems vague to me about it, in particular  reading the following `The accessor template parameter IsPlaceholder is allowed to be specified, but it has no bearing on whether the accessor instance is a placeholder. This is determined solely by the constructor used to create the instance.` (taken from [here](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_placeholder_template_parameter)).

I also did a test compile run with the latest dpcpp nightly and it doesn't flag it as deprecated, so I think we can still leave it at least for now.

Reply:
OK, lets leave it for now. 
I think the intention is old-style placeholder accessors don't get broken, but also that they don't actually do anything.